diff --git a/gdk-pixbuf/gdk-pixbuf-io.c b/gdk-pixbuf/gdk-pixbuf-io.c
index 40eb920..e8379e9 100644
--- a/gdk-pixbuf/gdk-pixbuf-io.c
+++ b/gdk-pixbuf/gdk-pixbuf-io.c
@@ -933,7 +933,10 @@ _gdk_pixbuf_get_module (guchar *buffer, guint size,
 		}
         }
         g_free (mime_type);
-#else
+#endif
+	if (selected != NULL)
+		return selected;
+
         gint score, best = 0;
 
         for (modules = get_file_formats (); modules; modules = g_slist_next (modules)) {
@@ -950,7 +953,7 @@ _gdk_pixbuf_get_module (guchar *buffer, guint size,
                 if (score >= 100) 
                         break;
         }
-#endif
+
 
         if (selected != NULL)
                 return selected;
